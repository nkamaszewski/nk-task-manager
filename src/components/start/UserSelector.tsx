import React from "react";
import { USERS } from "../../helpers/constans";
import { User } from "../../types/types";
import UserSelectorStyle from "../../styles/UserSelectorStyle";

interface Props {
  selectedUserId: number;
  setSelectedUserId: Function;
}

const UserSelector = ({ selectedUserId, setSelectedUserId }: Props) => {
  return (
    <UserSelectorStyle>
      <p className="title">Select mock user:</p>
      {USERS.map(({ name, id }: User) => (
        <label key={id} className={name} title={`userId: ${id}`}>
          <input
            className="radio"
            type="radio"
            value={id}
            checked={id === selectedUserId}
            onChange={() => setSelectedUserId(id)}
          />
          {`${name}`}
        </label>
      ))}
    </UserSelectorStyle>
  );
};

export default UserSelector;
