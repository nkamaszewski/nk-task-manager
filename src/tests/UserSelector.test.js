import React from "react";
import renderer from "react-test-renderer";
import UserSelector from "../components/start/UserSelector";

it("UserSelector renders correctly", () => {
  const tree = renderer.create(<UserSelector />).toJSON();
  expect(tree).toMatchSnapshot();
});
