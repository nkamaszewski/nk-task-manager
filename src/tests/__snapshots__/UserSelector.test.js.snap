// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSelector renders correctly 1`] = `
<div
  className="sc-bdVaJa jdJPjS"
>
  <p
    className="title"
  >
    Select mock user:
  </p>
  <label
    className="Alice"
    title="userId: 1"
  >
    <input
      checked={false}
      className="radio"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    Alice
  </label>
  <label
    className="Bob"
    title="userId: 2"
  >
    <input
      checked={false}
      className="radio"
      onChange={[Function]}
      type="radio"
      value={2}
    />
    Bob
  </label>
  <label
    className="Frank"
    title="userId: 3"
  >
    <input
      checked={false}
      className="radio"
      onChange={[Function]}
      type="radio"
      value={3}
    />
    Frank
  </label>
</div>
`;
